### ISSUE TITLE
Please update python version and fix ssl issue

### ISSUE THREAD

**Issue by stefaneidelloth** [qgis/QGIS#50993]
### Feature description

I would like to use QGis with Python 3.10.5. However, the python version that comes with QGis is outdated 9.3.5 and has an issue with SSH. 
https://stackoverflow.com/questions/60290795/ssl-module-in-python-is-not-available-qgis

(The suggested solution to copy the ddl files works to fix the SSH issue.) 

=> Please update Python

=> It would be great if QGIS could be installed as a pip package, ensuring version compatibility. 

I would like to use QGis from my own Python instance, where I already installed some dependencies. 
If you force me to use the old Python version that comes with QGis, I need to reinstall dependencies and
use older versions. Furthermore, installing dependencies does not work without manually fixing the ssh issue first.

**Comment by nicogodet** [comment]
Regarding python upgrade:
If your are on Windows -> [OSGeo4W Issue Tracker](https://trac.osgeo.org/osgeo4w/query?status=assigned&status=closed&status=new&status=reopened&col=id&col=summary&col=status&col=type&col=priority&col=component&col=resolution&col=time&col=changetime&col=reporter&desc=1&order=id&report=12)
If you are on MacOS -> [MacOS Packager Issue Tracker](https://github.com/qgis/QGIS-Mac-Packager/issues)
As I believe you are on one of these 2 plateforms, I will not explain for Linux.

Conda is working on a QGIS package AFAIK : https://anaconda.org/conda-forge/qgis

**Comment by nicogodet** [comment]
But I think this FR can be closed as it's not directly related to QGIS it self but packaging on each OS.

**Comment by stefaneidelloth** [comment]
Is it possible to install qgis using pip, somehow referencing this Git repository?

**Comment by nicogodet** [comment]
> Is it possible to install qgis using pip, somehow referencing this Git repository?

Not possible ATM

**Comment by stefaneidelloth** [comment]
Thank you. I did not find out how I can create a new issue ticket on the referenced [OSGeo4W Issue Tracker](https://trac.osgeo.org/osgeo4w/query?status=assigned&status=closed&status=new&status=reopened&col=id&col=summary&col=status&col=type&col=priority&col=component&col=resolution&col=time&col=changetime&col=reporter&desc=1&order=id&report=12)

**Comment by jef-n** [comment]
OSGeo4W should be fine.

```
C:\osgeo4w>python3
Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import ssl
>>> ssl.OPENSSL_VERSION
'OpenSSL 1.1.1s  1 Nov 2022'
```

**Comment by nicogodet** [comment]
> Seems they do not want to get feedback.

Easy on criticism...
You need an OSGeo account to create a ticket on OSGeo4W tracker.

> Only authenticated (logged in) users can submit and edit tickets, and modify the wiki. Use your [​OSGeo userid/password](https://www.osgeo.org/osgeo_userid) to login.

It seems you didn't search a lot before posting...

**Comment by stefaneidelloth** [comment]
>It seems you didn't search a lot before posting...
Exactly, if there is no "register" button, I need to search a lot... create extra account... seriously? 
If they really wanted to get feedback they would make it easier, e.g using GitHub like this project. 
Maybe they have enough open issues without feedback.  
Nevertheless thank you for your efforts and the info.

**Comment by jef-n** [comment]
> sqlite.dll is also missing. If somebody else runs in the same issue: copy all dll files.

sqlite3.dll is in %OSGEO4W_ROOT%\bin

**Comment by stefaneidelloth** [comment]
>sqlite3.dll is in %OSGEO4W_ROOT%\bin

Exactly, as written in 
https://stackoverflow.com/questions/60290795/ssl-module-in-python-is-not-available-qgis
needs to be copied to
%OSGEO4W_ROOT%\apps\Python3x\DLLs

**Comment by jef-n** [comment]
> Exactly, as suggested in https://stackoverflow.com/questions/60290795/ssl-module-in-python-is-not-available-qgis needs 
to be copied to %OSGEO4W_ROOT%\apps\Python3x\DLLs

Why?

**Comment by stefaneidelloth** [comment]
![image](https://user-images.githubusercontent.com/12675339/203814775-8ab702eb-8ae9-4b92-9894-8456f8539c53.png)

![image](https://user-images.githubusercontent.com/12675339/203815577-f86246ac-8f2c-44d1-9ad8-7f97dcca03d0.png)

**Comment by jef-n** [comment]
Use OSGeo4W shell:

```
run o-help for a list of available commands
C:\OSGeo4W>pip install iface
Collecting iface
  Downloading iface-0.1.2.tar.gz (9.0 kB)
  Preparing metadata (setup.py) ... done
Installing collected packages: iface
  DEPRECATION: iface is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for iface ... done
Successfully installed iface-0.1.2

[notice] A new release of pip available: 22.3 -> 22.3.1
[notice] To update, run: python3.exe -m pip install --upgrade pip

C:\OSGeo4W>
```
